java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -522,7 +540,137 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
+   *                       `col_narep` is valid.
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenate(ColumnView[] columns,
+      ColumnView sep_col, Scalar separator_narep, Scalar col_narep, boolean separate_nulls) {
+    assert columns.length >= 1 : ".stringConcatenate() operation requires at least 1 column";
+    assert separator_narep != null : "separator narep scalar provided may not be null";
+    assert col_narep != null : "column narep scalar provided may not be null";
+    assert separator_narep.getType().equals(DType.STRING) : "separator naprep scalar must be a string scalar";
+    assert col_narep.getType().equals(DType.STRING) : "column narep scalar must be a string scalar";
+
+    long[] column_views = new long[columns.length];
+    for(int i = 0; i < columns.length; i++) {
+      assert columns[i] != null : "Column vectors passed may not be null";
+      column_views[i] = columns[i].getNativeView();
+    }
+
+    return new ColumnVector(stringConcatenationSepCol(column_views, sep_col.getNativeView(),
+      separator_narep.getScalarHandle(), col_narep.getScalarHandle(), separate_nulls));
+  }
+
+  /**
+   * Given a lists column of strings (each row is a list of strings), concatenates the strings
+   * within each row and returns a single strings column result. Each new string is created by
+   * concatenating the strings from the same row (same list element) delimited by the separator
+   * provided. This version of the function relaces nulls with empty string and returns null
+   * for empty list.
+   * @param list_column column containing lists of strings to concatenate.
+   * @param sep_col strings column that provides separators for concatenation.
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenateListElements(ColumnView list_column,
+      ColumnView sep_col) {
+    try (Scalar nullString = Scalar.fromString(null);
+         Scalar emptyString = Scalar.fromString("")) {
+      return stringConcatenateListElements(list_column, sep_col, nullString, emptyString,
+          false, false);
+    }
+  }
+
+  /**
+   * Given a lists column of strings (each row is a list of strings), concatenates the strings
+   * within each row and returns a single strings column result.
+   * Each new string is created by concatenating the strings from the same row (same list element)
+   * delimited by the row separator provided in the sep_colstrings column.
+   * @param list_column column containing lists of strings to concatenate.
+   * @param sep_col strings column that provides separators for concatenation.
+   * @param separator_narep string that should be used to replace null separator, default is an
+   *                        invalid-scalar denoting that rows containing null separator will
```

Comment:
Confused by this mention of default since it is a required parameter and thus there's no default.  Should this simply mention how an invalid scalar will be handled?

Reply:
copy and paste error from the cudf api, I'll fix up and go through rest, was trying to update with the new api's